sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java

```diff
@@ -2610,6 +2622,33 @@ interface WithDeleteOptions {
              */
             Update withDataDisksDeleteOptions(DeleteOptions deleteOptions, Integer... luns);
 
+            /**
+             * Update delete option for the primary public ip address of the primary network interface.
+             *
+             * @param deleteOptions what happens to the public IP address when the VM using it is deleted
+             * @return the next stage of the update
+             */
+            Update withPrimaryNicPrimaryPipDeleteOption(DeleteOptions deleteOptions);
+
+            /**
+             * Update delete option for the primary public ip address of the specified network interface.
+             *
+             * @param deleteOptions what happens to the public IP address when the VM using it is deleted
+             * @param nicIds resource Ids of the specified network interface
+             * @return the next stage of the update
+             */
+            Update withSpecifiedNicPrimaryPIpDeleteOption(DeleteOptions deleteOptions, String... nicIds);
+
+            /**
+             * Update delete option for the specified public ip address of the specified network interface.
+             *
+             * @param deleteOptions what happens to the public IP address when the VM using it is deleted
+             * @param nicId resource id of the specified network interfac
+             * @param ipConfigIds resource Ids of specified network interface ip configs
+             * @return the next stage of the update
+             */
+            Update withSpecifiedNicSpecifiedPIpDeleteOption(DeleteOptions deleteOptions, String nicId, String... ipConfigIds);
```

Comment:
Not sure whether user really needs these APIs.

IMHO, two are needed:
1. `withPublicIPAddressDeleteOptions(DeleteOptions deleteOptions)` to update all attached PublicIpAddress'es deleteOption
2. `withPublicIPAddressDeleteOption(PublicIpAddress(/String) publicIpAddress(id))`, like what user specifies in the creation flow.